framework/src/utils/RankThreeTensor.C

```diff
@@ -23,6 +23,11 @@
 #include <iomanip>
 #include <ostream>
 
+#define r3i(i, j, k) ((i * LIBMESH_DIM + j) * LIBMESH_DIM + k)
+#define N LIBMESH_DIM
+#define N2 (LIBMESH_DIM * LIBMESH_DIM)
+#define N3 (LIBMESH_DIM * LIBMESH_DIM * LIBMESH_DIM)
```

Comment:
Sure, but `constexpr` and inline functions will behave the same way in modern compilers.